### PR TITLE
Text summarization

### PR DESCRIPTION
**Pull request by Vijay-Nirmal**

### CODE REVIEW

File: LanguageEngine/Summarizer/Gensim.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+warnings.filterwarnings(action='ignore', category=UserWarning, module='gensim')
+
+from gensim.summarization.summarizer import summarize
+from gensim.summarization import keywords
+from bs4 import BeautifulSoup
+from urllib.request import urlopen
+import re
+
+def removeRefNo(inputString):
+    """Removes reference numbers from the string
```
Comment (AakashSasikumar) [inline LanguageEngine/Summarizer/Gensim.py]: mention that it is the wikipedia reference numbers just to make it clear
```suggestion
    """Removes reference numbers from the strings obtained from wikipedia
```

File: LanguageEngine/Summarizer/Gensim.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    
+    Parameters
+    ----------
+    inputString : String
+        Input string
+
+    Returns
+    -------
+    inputString : String
+        Reference number removed string
```
Comment (AakashSasikumar) [inline LanguageEngine/Summarizer/Gensim.py]: same thing applies here

File: LanguageEngine/Summarizer/Gensim.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    
+    """
+
+    return re.sub(r"[\[].*?[\]]", "", inputString)
+ 
+def getOnlyText(url, noOfPar = 0):
```
Comment (AakashSasikumar) [inline LanguageEngine/Summarizer/Gensim.py]:
```suggestion
def getOnlyText(url, noOfParagraphs = 0):
```

Comment (AakashSasikumar) [reply LanguageEngine/Summarizer/Gensim.py]: for code readability

File: LanguageEngine/Summarizer/Gensim.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        text = ' '.join(map(lambda p: p.text, soup.find_all('p')))
+    else:
+        for i, para in enumerate(soup.find_all('p')):
+            if i >= noOfPar:
+                break
+            text += " " + para
+    
+    return text.strip()
+
+def GetSummaryFromURI(url, wordCount = 50, noOfPar = 0):
```
Comment (AakashSasikumar) [inline LanguageEngine/Summarizer/Gensim.py]: lower case
```suggestion
def getSummaryFromURI(url, wordCount = 50, noOfPar = 0):
```

File: LanguageEngine/Summarizer/Gensim.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+    Returns
+    -------
+    inputString : String
+        Summarized output
+    
+    """
+    return removeRefNo(summarize(str(getOnlyText(url, noOfPar)), word_count = wordCount))
+
+def GetSummary(inputString, wordCount = 50):
```
Comment (AakashSasikumar) [inline LanguageEngine/Summarizer/Gensim.py]: lower case start
```suggestion
def getSummary(inputString, wordCount = 50):
```

File: LanguageEngine/Summarizer/Gensim.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
+import warnings
+warnings.filterwarnings(action='ignore', category=UserWarning, module='gensim')
+
+from gensim.summarization.summarizer import summarize
+from gensim.summarization import keywords
+from bs4 import BeautifulSoup
+from urllib.request import urlopen
+import re
+
+def removeRefNo(inputString):
```
Comment (AakashSasikumar) [inline LanguageEngine/Summarizer/Gensim.py]: improve readability
```suggestion
def removeReferenceNumbers(inputString):
```

File: LanguageEngine/Summarizer/NLTKFrequencySummarizer.py [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    inputString : `String`
+        Reference number removed string
+    
+    """
+
+    return re.sub(r"[\[].*?[\]]", "", inputString)
+
+def rank(ranking, n):
+    """ Return the first n sentences with highest ranking
+
```
Comment (AakashSasikumar) [inline LanguageEngine/Summarizer/NLTKFrequencySummarizer.py]: mention the functions of these parameters

File: LanguageEngine/Summarizer/NLTKFrequencySummarizer.py [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+        text = ' '.join(map(lambda p: p.text, soup.find_all('p')))
+    else:
+        for i, para in enumerate(soup.find_all('p')):
+            if i >= noOfPar:
+                break
+            text += " " + para
+    
+    return text.strip()
+
+def GetSummaryFromURI(url, sentencesCount = 5, noOfPar = 0):
```
Comment (AakashSasikumar) [inline LanguageEngine/Summarizer/NLTKFrequencySummarizer.py]: lower case start

File: LanguageEngine/Summarizer/NLTKFrequencySummarizer.py [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+    Returns
+    -------
+    inputString : String
+        Summarized output
+    
+    """
+    return summarize(getOnlyText(url, noOfPar), sentencesCount)
+
+def GetSummary(inputString, sentencesCount = 5):
```
Comment (AakashSasikumar) [inline LanguageEngine/Summarizer/NLTKFrequencySummarizer.py]: lower case start

File: README.md [hunk 10]
```diff
@@ -13,6 +13,7 @@ pip install tflearn
 pip install nltk
 pip install beautifulsoup4
 pip install lxml
+pip install --upgrade gensim
```
Comment (AakashSasikumar) [inline README.md]: it can just be this if people haven't already installed
```suggestion
pip install gensim
```

Comment (Vijay-Nirmal) [reply README.md]: `pip install gensim` didn't work

File: LanguageEngine/Summarizer/Gensim.py (hunk 2 above)
Comment (Vijay-Nirmal) [reply LanguageEngine/Summarizer/Gensim.py]: What same?

**Comment by AakashSasikumar** [comment]
looks good

**Pull request merged**